core/src/main/scala/io/chrisdavenport/shellfish/syntax/path/package.scala

```diff
@@ -0,0 +1,374 @@
+/*
+ * Copyright (c) 2024 Typelevel
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package io.chrisdavenport.shellfish
+package syntax
+
+import cats.effect.IO
+
+import fs2.io.file.*
+
+import scodec.bits.ByteVector
+import scodec.Codec
+
+import scala.concurrent.duration.FiniteDuration
+
+package object path {
+
+  implicit class ReadWriteOps(val path: Path) extends AnyVal {
```

Comment:
Overall design wise, we should either pick methods on an object (`FilesOs`) or extension methods on relevant types like `Path`. Don't give users multiple ways to do the same thing.

Reply:
@mpilquist I think this point warrants some more discussion. It got mentioned that @TonioGela wanted both designs possible.

While I agree, that for scripting purposes a single API makes more sense. Let's see what others say and try to make a decision based off that?

Reply:
Oh I interpreted that way because of an example he gave me 😂 but it can be totally the case Tonio also prefers only one. 

Honestly I tried to make both as Thanh prefer the one calling the object directly and Tonio suggested using the one with extension methods so I think it would be nice to let the user decide which one they want to use 😄. I also did it that way because in Cats you can also call both the API using the syntax or the typeclass instance (and if we end up naming this project Cats OS I think that be more consistent). 

At the other end I also get the disadvantages of letting the user choose more than one API (confusion regarding which one of the two is the default one, problems mixing both APIs in the same script, more code to maintain, etcetera)... So as Rishad said we should discuss this a bit further. If we end up choosing only one IMO I prefer the extension methods one as if we write `FilesOs.writeUtf8Lines(path) ... ` might as well just `Files[IO].writeUtf8Lines(path) ...` 😅

Reply:
My catch on the lib is that its purpose is exactly to offer alternative ways to do the same thing, so I'm in favor of keeping both the static calls like `FilesOs.writeLines(path, lines)` and `path.writeLines(lines)`.